tests/test_fpca.py

```diff
@@ -0,0 +1,82 @@
+import unittest
+
+import numpy as np
+from skfda import FDataGrid, FDataBasis
+from skfda.representation.basis import Fourier
+from skfda.preprocessing.dim_reduction.projection import FPCABasis, FPCAGrid
+from skfda.datasets import fetch_weather
+
+
+class FPCATestCase(unittest.TestCase):
+
+    def test_basis_fpca_fit_attributes(self):
+        fpca = FPCABasis()
+        with self.assertRaises(AttributeError):
+            fpca.fit(None)
+
+        basis = Fourier(n_basis=1)
+        # check that if n_components is bigger than the number of samples then
+        # an exception should be thrown
+        fd = FDataBasis(basis, [[0.9]])
+        with self.assertRaises(AttributeError):
+            fpca.fit(fd)
+
+        # check that n_components must be smaller than the number of elements
+        # of target basis
+        fd = FDataBasis(basis, [[0.9], [0.7], [0.5]])
+        with self.assertRaises(AttributeError):
+            fpca.fit(fd)
+
+    def test_discretized_fpca_fit_attributes(self):
+        fpca = FPCAGrid()
+        with self.assertRaises(AttributeError):
+            fpca.fit(None)
+
+        # check that if n_components is bigger than the number of samples then
+        # an exception should be thrown
+        fd = FDataGrid([[0.5], [0.1]], sample_points=[0])
+        with self.assertRaises(AttributeError):
+            fpca.fit(fd)
+
+        # check that n_components must be smaller than the number of attributes
+        # in the FDataGrid object
+        fd = FDataGrid([[0.9], [0.7], [0.5]], sample_points=[0])
+        with self.assertRaises(AttributeError):
+            fpca.fit(fd)
+
+    def test_basis_fpca_fit_result(self):
```

Comment:
What about a similar test for the discretized form?

Reply:
Done, it took me some time to study fda.usc, but better late than never right